atmcfg/atmcfg.go

```diff
@@ -419,3 +419,51 @@ func Restart(out *opsmngr.AutomationConfig, name string) {
 	restartByReplicaSetName(out, name, lastRestart)
 	restartByShardName(out, name, lastRestart)
 }
+
+// ReclaimFreeSpace sets all process of a cluster to reclaim free space
+func ReclaimFreeSpace(out *opsmngr.AutomationConfig, name string) {
+	// This value may not be present and is mandatory
+	if out.Auth.DeploymentAuthMechanisms == nil {
+		out.Auth.DeploymentAuthMechanisms = make([]string, 0)
+	}
+	lastCompact := time.Now().Format(time.RFC3339)
+	reclaimByReplicaSetName(out, name, lastCompact)
+	reclaimByShardName(out, name, lastCompact)
+}
```

Comment:
[q] Where are we going to use this function in mongocli?

Reply:
https://feedback.mongodb.com/forums/930808-mongodb-cli/suggestions/41674054-allow-to-perform-rolling-compact-via-restapi

Reply:
Also I'm trying to confirm if it makes sense to set this for mongos as I think maybe not